File: app/src/main/java/com/aayush/filmipedia/util/viewpager/NoSwipePager.java
package com.aayush.filmipedia.util.viewpager;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.viewpager.widget.ViewPager;

public class NoSwipePager extends ViewPager {
    private boolean enabled;

    public NoSwipePager(Context context, AttributeSet attrs) {
        super(context, attrs);
        enabled = true;
    }

    @Override
    public boolean onTouchEvent(MotionEvent ev) {
        if (enabled) {
            performClick();
            return super.onTouchEvent(ev);
        }
        return false;
    }

    @Override
    public boolean performClick() {
        return super.performClick();
    }

    @Override
    public boolean onInterceptTouchEvent(MotionEvent ev) {
        if (enabled) {
            return super.onInterceptTouchEvent(ev);
        }
        return false;
    }

    public void setPagingEnabled(boolean enabled) {
        this.enabled = enabled;
    }
}

File: app/src/main/java/com/aayush/filmipedia/util/viewpager/NoSwipePager.kt
package com.aayush.filmipedia.util.viewpager

import android.content.Context
import android.util.AttributeSet
import android.view.MotionEvent

import androidx.viewpager.widget.ViewPager

class NoSwipePager(context: Context, attrs: AttributeSet): ViewPager(context, attrs) {
    var isSwipeEnabled = true

    override fun onTouchEvent(ev: MotionEvent) =
        if (isSwipeEnabled) {
            performClick()
            super.onTouchEvent(ev)
        } else false

    override fun performClick() = super.performClick()

    override fun onInterceptTouchEvent(ev: MotionEvent) =
        if (isSwipeEnabled) {
            super.onInterceptTouchEvent(ev)
        } else false
}